src/shared/tpm2-util.c

```diff
@@ -717,9 +739,24 @@ static int tpm2_make_pcr_session(
         return r;
 }
 
+static void hash_pin(const char *pin, uint8_t ret_digest[static SHA256_DIGEST_SIZE]) {
+        struct sha256_ctx hash;
+        size_t pin_len;
+
+        assert(pin);
+        pin_len = strlen(pin);
+
+        sha256_init_ctx(&hash);
+        sha256_process_bytes(pin, pin_len, &hash);
```

Comment:
doesn't really matter, but I'd just pass `strlen(pin)` as second parameter directly here, there's no real benefit in using a temporary variable `pin_len`...